execute/table/static/static.go

```diff
@@ -533,15 +595,109 @@ func mustTimeValue(v interface{}, offset int64, unit time.Duration) (t int64, ab
 	}
 }
 
-// Extend will copy the spec from the table and add or override
-// any columns for the extended table.
-func (s Table) Extend(table Table) Table {
-	ns := make(Table, len(s)+len(table))
-	for label, column := range s {
-		ns[label] = column
+// TableBuilder is used to construct a set of Tables.
+type TableBuilder interface {
+	// Build will construct a set of tables using the
+	// template as input.
+	//
+	// The template is a pointer as a builder is allowed
+	// to modify the template. For implementors, the
+	// template pointer must be non-nil.
+	Build(template *[]Column) []flux.Table
+}
+
```

Comment:
What happened to `TableMatrix`? I thought that would be in here somewhere.

Reply:
It should be in there towards the bottom.

Reply:
Ah, so it is, I was reviewing when you force pushed and it disappeared on me.